web/src/notifications.js

```diff
@@ -241,7 +241,12 @@ export function process_notification(notification) {
     const $content = $("<div>").html(message.content);
     ui.replace_emoji_with_text($content);
     spoilers.hide_spoilers_in_notification($content);
-    content = $content.text();
+
+    if ($content.text().trim() === "") {
+        content = $t({defaultMessage: "(attached file)"});
```

Comment:
`trim()` removes a lot of characters from the text, we cannot assume it is a file like this. We should additionally check `message_has_attachment`.

Let's add a test case for this too.

Reply:
@amanagr I tried to add the test but it seems like the body of the notification is always set to `never-been-set` when we are creating one from tests. I tracked this behavior to `FakeElement` of `zjquery_element.js`. I was not really able to figure out a solution. Can you guide me here?

Reply:
You can look at other examples in `notifications.test.js`. I am not sure what you mean.

Reply:
@amanagr What I am trying to communicate is, we're setting `$content` [here](https://github.com/zulip/zulip/blob/main/web/src/notifications.js#L241) like `const $content = $("<div>").html(message.content);` but the `$content.text()` does not return the expected result in the test environment and rather return `never-been-set` but it works just fine in `run-dev`. Also, there are no tests that are similar that I can take as a reference to write a test for this.

Is it so that jquery behaves differently in the test environment?

Reply:
Yes, see zjquery.test.js. I think you can set a custom tests in tests using `$container.text("`") and then verify that you `if` condition works correctly.

Reply:
The following is the response that is returned when we call `notifications.get_notifications();`
```Javascript
{
  obj: StubNotification {
    icon: 'http://zulip.zulipdev.com/url?s=50',
    body: 'never-been-set',
    tag: 1000
  },
  msg_count: 1,
  message_id: 1000
}
```
`body` is the attribute that will help us verify that a notification is showing content that is expected.

The `process_notifications` is the function that is responsible to create that object and it uses `$content.text()` [here](https://github.com/zulip/zulip/blob/main/web/src/notifications.js#L244) to set the body field. I think the `zjquery` is not able to process `$content.html("whatever")` beyond of course updating the `html` of `$content` and hence not able to return `$content.text()` properly.

This is what I'm doing,
```javascript
test("attachement_notifications", () => {
    // zjquery element.
    $("<div>").set_find_results(".emoji", {replaceWith() {}});

    // Notifications API stub
    class StubNotification {
        constructor(title, {icon, body, tag}) {
            this.icon = icon;
            this.body = body;
            this.tag = tag;
        }
    }

    notifications.set_notification_api(StubNotification);

    const message_1 = {
        id: 1000,
        content: '<a href="http://example.com/whatever.png"></a>',
        avatar_url: "url",
        type: "private",
        display_recipient: {},
        display_reply_to: "whatever",
    };

    notifications.process_notification({message: message_1, desktop_notify: true});

    const n = notifications.get_notifications();
    console.log(n); // We will be checking the response here.
});
```